src/features/bookings/components/BookingDetailsTicketContent.tsx

```diff
@@ -159,66 +153,66 @@ const NoTicket = () => {
     return <QrCodeView qrCodeData={booking.qrCodeData} />
 
+  if (!withdrawalType) {
+    return null
+  }
+
   if (withdrawalType === WithdrawalTypeEnum.no_ticket) {
     return <NoTicket />
   }
+  const { beginningDatetime } = booking.stock
 
-  if (
-    withdrawalType === WithdrawalTypeEnum.on_site ||
-    withdrawalType === WithdrawalTypeEnum.by_email
-  ) {
-    const { beginningDatetime } = booking.stock
-
-    if (beginningDatetime && withdrawalType === WithdrawalTypeEnum.by_email) {
-      // Calculation approximate date send e-mail
-      const nbDays = withdrawalDelay / 60 / 60 / 24
-      const dateSendEmail = addDays(new Date(beginningDatetime), -nbDays)
-      const today = new Date()
-      const startOfferDate = new Date(beginningDatetime)
-
-      if (isSameDay(startOfferDate, today) || today > dateSendEmail) {
-        return <TicketEmailSent offerDate={startOfferDate} />
-      }
+  if (beginningDatetime && withdrawalType === WithdrawalTypeEnum.by_email) {
+    // Calculation approximate date send e-mail
+    const nbDays = withdrawalDelay / 60 / 60 / 24
+    const dateSendEmail = addDays(new Date(beginningDatetime), -nbDays)
+    const today = new Date()
+    const startOfferDate = new Date(beginningDatetime)
+
+    if (isSameDay(startOfferDate, today) || today > dateSendEmail) {
+      return <TicketEmailSent offerDate={startOfferDate} />
     }
+  }
+
+  const startMessage = getStartMessage(withdrawalType)
+  const delayMessage = getDelayMessage(withdrawalDelay)
+  const endMessage = t`avant le début de l’événement.`
 
-    const startMessage =
-      (withdrawalType === WithdrawalTypeEnum.on_site
-        ? t`Présente le code ci-dessus sur place`
-        : t`Tu vas recevoir ton billet par e-mail`) + ' '
-    const delayMessage = withdrawalDelay > 0 ? `${formatSecondsToString(withdrawalDelay)} ` : null
-    const endMessage = t`avant le début de l’événement.`
-
-    return (
-      <React.Fragment>
-        {withdrawalType === WithdrawalTypeEnum.by_email && (
-          <IconContainer>
-            <BicolorEmailSent />
-          </IconContainer>
+  return (
+    <React.Fragment>
+      {withdrawalType === WithdrawalTypeEnum.by_email && (
+        <IconContainer>
+          <BicolorEmailSent />
+        </IconContainer>
+      )}
+
+      <TicketInfo testID="withdrawal-info">
+        {startMessage}
+        {!!delayMessage && (
+          <TicketCollectDelay testID="withdrawal-info-delay">{delayMessage}</TicketCollectDelay>
         )}
+        {endMessage}
+      </TicketInfo>
+    </React.Fragment>
+  )
+}
 
-        <TicketInfo testID="withdrawal-info">
-          {startMessage}
-          {!!delayMessage && (
-            <TicketCollectDelay testID="withdrawal-info-delay">{delayMessage}</TicketCollectDelay>
-          )}
-          {endMessage}
-        </TicketInfo>
-      </React.Fragment>
-    )
-  }
+const getStartMessage = (withdrawalType: string): string => {
```

Comment:
je pense que `string` est trop permissive et mensongé, on pourrait mettre

```suggestion
const getStartMessage = (withdrawalType: WithdrawalTypeEnum): string => {
```